internal/store/store.go

```diff
@@ -89,235 +106,268 @@ func customCATransport(ca []byte) http.RoundTripper {
 	}
+	options = append(options, NetworkPresets(c))
+	return Options(options...)
+}
 
-	client, err := defaultClient(c)
-	if err != nil {
-		return nil, err
+// PublicUnauthenticatedPreset default settings when connecting to the public API without authentication.
+func PublicUnauthenticatedPreset(c Config) Option {
+	options := []Option{Service(c.Service())}
+	if configURL := c.OpsManagerURL(); configURL != "" {
+		options = append(options, WithBaseURL(configURL), WithPublicPathBaseURL())
 	}
+	options = append(options, NetworkPresets(c))
+	return Options(options...)
+}
 
-	if err2 := s.setAtlasClient(client); err2 != nil {
-		return nil, err2
+// PrivateAuthenticatedPreset default settings when connecting to the private API with authentication.
+func PrivateAuthenticatedPreset(c Config) Option {
+	options := []Option{Service(c.Service()), WithAuthentication(c)}
+	if configURL := c.OpsManagerURL(); configURL != "" {
+		options = append(options, WithBaseURL(configURL))
 	}
-
-	return s, err
+	options = append(options, NetworkPresets(c))
+	return Options(options...)
 }
 
-// NewPrivate gets the appropriate client for the atlas private api with authentication
-func NewPrivate(c Config) (*Store, error) {
-	s := new(Store)
-	s.service = c.Service()
-	if s.service != config.CloudService {
-		return nil, fmt.Errorf("unsupported service: %s", s.service)
+// PrivateUnauthenticatedPreset default settings when connecting to the private API without authentication.
+func PrivateUnauthenticatedPreset(c Config) Option {
+	options := []Option{Service(c.Service())}
+	if configURL := c.OpsManagerURL(); configURL != "" {
+		options = append(options, WithBaseURL(configURL))
 	}
-	s.baseURL = atlas.CloudURL
+	options = append(options, NetworkPresets(c))
+	return Options(options...)
+}
 
-	if configURL := c.OpsManagerURL(); configURL != "" {
-		s.baseURL = configURL
+// New return a new Store based on the given list of Option.
+//
+// Usage:
+//
+//	// get a new client for Atlas
+//	store := store.New(WithService("cloud"))
+//
+//	// get a new client based on the config
+//	store := store.New(PublicAuthenticatedPreset(config))
+func New(opts ...Option) (*Store, error) {
+	store := new(Store)
+
+	// apply the list of options to Server
+	for _, opt := range opts {
+		if err := opt(store); err != nil {
+			return nil, err
+		}
 	}
 
-	client, err := authenticatedClient(c)
+	httpTransport, err := store.transport()
 	if err != nil {
 		return nil, err
 	}
-
-	if err2 := s.setAtlasClient(client); err2 != nil {
-		return nil, err2
+	client, err := store.httpClient(httpTransport)
+	if err != nil {
+		return nil, err
 	}
 
-	return s, err
-}
-
-func (s *Store) setAtlasClient(client *http.Client) error {
-	opts := make([]atlas.ClientOpt, 0)
-	if s.baseURL != "" {
-		opts = append(opts, atlas.SetBaseURL(s.baseURL))
+	switch store.service {
+	case config.CloudService:
+		err = withAtlasClient(client)(store)
```

Comment:
[nit] I personally don't like the name `withAtlasClient`, maybe `AtlasClientOption` would be better

Reply:
why not? the reason behind the `with` is that you want a `Store` `with` an atlas client smae for all the `with` around